Understanding metrics topic (#3426)
centreon/centreon-documentation: i18n/fr/docusaurus-plugin-content-docs/version-24.04/monitoring/metrics.md

---
id: metrics
title: Comprendre les métriques
---

## Qu'est-ce qu'une métrique ?

Une métrique, ou "donnée de performance", est un élément d'un service. Un service peut comprendre plusieurs métriques, ou aucune.

Les métriques permettent d’obtenir des graphiques et de positionner des seuils pour être alerté.

### Statut des métriques et statut du service

> Lorsqu’un service comprend plusieurs métriques, le statut du service est celui de la plus mauvaise métrique. L'ordre de priorité est le suivant : OK, INCONNU, ALERTE, CRITIQUE.

Reviewer: ```suggestion
Lorsqu’un service comprend plusieurs métriques, le statut du service est le plus mauvais parmi toutes les métriques. L'ordre de priorité est le suivant : OK, INCONNU, ALERTE, CRITIQUE.
```

Si une métrique n'est pas récupérée, son statut est considéré comme INCONNU et impactera le statut du service associé.

### Services sans métriques

> Certains services n'ont aucune métrique associée : il s'agit de services retournant des statuts et non des valeurs numériques. Typiquement, des contrôles sur des équipements matériels. Le statut que prend le service est directement défini dans le code de chaque plugin, en fonction du code retour de l'équipement.

Reviewer: ```suggestion
Certains services n'ont aucune métrique associée : il s'agit de services contrôlant des états et non des valeurs numériques. Typiquement, des contrôles sur des équipements matériels. Le statut que prend le service est directement défini dans le code de chaque plugin, en fonction de la réponse de l'équipement.
```

## Où voir les métriques d'un service dans Centreon ?

À la page **Statut des ressources**, dans le panneau de détails d'un service, la section **Données de performance** donne les valeurs détaillées de toutes les métriques associées à ce service, pour le dernier contrôle effectué.

Exemple : Nous supervisons un collecteur Centreon (pour ce service, via le connecteur [Linux SNMP](https://docs.centreon.com/pp/integrations/plugin-packs/procedures/operatingsystems-linux-snmp/#collected-metrics--status)). Dans l'image suivante, le service **Memory** retourne 6 métriques : **used**, **free**, **used_prct**, **buffer**, **cached** et **shared**.

![image](../assets/monitoring/metrics.png)

Vous pouvez également :

* Voir l'évolution de toutes les métriques d'un service dans l'onglet **Graphique** du panneau de détails du service.
* Représenter le dernière valeur d'une métrique dans un [tableau de bord](../alerts-notifications/dashboards.md), grâce au widget **Métrique unique**.

## Quel est le format d'une métrique ?

Dans tous les cas, le format des métriques telles que retournées par les plugins est le suivant :

```text

> 'nom_de_la_métrique'= valeur&unité;alerte;critique;min;max

Reviewer: & ?

```

* **valeur** : la valeur affichée dans le panneau de détails est celle retournée par le dernier contrôle effectué.

* **alerte/critique**: [seuils](#syntaxe-des-seuils-des-métriques) pour lesquels le statut de la métrique sera ALERTE ou CRITIQUE. Si aucun seuil n'est défini, le statut de la métrique sera toujours OK. La valeur de ces seuils est définie [dans une macro spécifique au service, ou bien dans la macro EXTRAOPTIONS du service](#comment-définir-des-seuils-pour-des-métriques-spécifiques).

>    > Il est possible de ne définir qu'un seuil CRITIQUE, sans passer par le stade ALERTE.

Reviewer: ```suggestion
   > Il est possible de ne définir qu'un seuil CRITIQUE, sans passer par le stade ALERTE et réciproquement.
```

> * **min/max** : Il ne s'agit pas de la valeur minimale/maximale constatée dans les données de performance, mais de la valeur minimale/maximale possible que peut prendre la métrique. Par exemple, le maximum d'une utilisation CPU sera toujours 100%. Ces valeurs sont définies [dans la macro EXTRAOPTIONS](#comment-définir-des-seuils-pour-des-métriques-spécifiques).

Reviewer: ```suggestion
* **min/max** : Il ne s'agit pas de la valeur minimale/maximale constatée dans les données de performance, mais de la valeur minimale/maximale possible que peut prendre la métrique. Par exemple, le maximum d'une utilisation CPU sera toujours 100%. Pour certains plugins ou modes, ces valeurs peuvent être redéfinies [dans la macro EXTRAOPTIONS](#comment-définir-des-seuils-pour-des-métriques-spécifiques).
```

>    * Les valeurs min/max servent à rendre l'affichage des graphes le plus pertinent possible : le min et le max seront les valeurs minimales/maximales sur l'axe des Y ou des X du graphique. Si aucun min et max ne sont définis, le graphique s'adaptera en hauteur en fonction de la plus petite et la plus grande valeur constatées. Si un graphique rassemble plusieurs métriques, définir le même min et max permet de toutes les représenter à la même échelle.

Reviewer: ```suggestion
   * Les valeurs min/max servent à rendre l'affichage des graphes le plus pertinent possible : le min et le max seront les valeurs minimales/maximales sur l'axe des Y du graphique. Si aucun min et max ne sont définis, le graphique s'adaptera en hauteur en fonction de la plus petite et la plus grande valeur constatées. Si plusieurs graphiques représentant le même service pour plusieurs hôtes sont affichés ensemble, définir le même min et max permet de toutes les représenter à la même échelle.
```

>    * Elles servent également à corriger certaines valeurs par défaut. Par exemple, si on contrôle le trafic réseau sur une interface, l'interface fournit son débit maximum théorique au plugin, mais celle-ci peut être changée à la main si le débit réel est limité par le fait d'être connecté à un autre équipement. Cela a notamment un impact sur les chiffres exploités dans les rapports de capacité dans [MBI](../reporting/introduction.md).

Reviewer: Ce ne sont pas les valeurs min/max qui permettent ça, mais les options qui les redéfinissent.

>    * Elles servent également à corriger certaines valeurs par défaut. Par exemple, si on contrôle le trafic réseau sur une interface, l'interface fournit son débit maximum théorique au plugin, mais celle-ci peut être changée à la main si le débit réel est limité par le fait d'être connecté à un autre équipement. Cela a notamment un impact sur les chiffres exploités dans les rapports de capacité dans [MBI](../reporting/introduction.md).

Reviewer: ```suggestion
   * Elles servent également à évaluer le niveau d'utilisation d'une ressource par rapport à sa capacité, d'où l'intérêtde pouvoir les redéfinir dans certains cas. Par exemple, si on contrôle le trafic réseau sur une interface, l'interface fournit son débit maximum théorique au plugin, mais celle-ci peut être changée à la main si le débit réel est limité par le fait d'être connecté à un autre équipement. Cela a notamment un impact sur les chiffres exploités dans les rapports de capacité dans [MBI](../reporting/introduction.md).
```

Dans tous les cas, le format des métriques telles que retournées par les plugins est le suivant :

```text
'nom_de_la_métrique'= valeur[unité];alerte;critique;min;max
```

* **valeur** : la valeur affichée dans le panneau de détails est celle retournée par le dernier contrôle effectué.

* **alerte/critique**: [seuils](#syntaxe-des-seuils-des-métriques) pour lesquels le statut de la métrique sera ALERTE ou CRITIQUE. Si aucun seuil n'est défini, le statut de la métrique sera toujours OK. La valeur de ces seuils est définie [dans une macro spécifique au service, ou bien dans la macro EXTRAOPTIONS du service](#comment-définir-des-seuils-pour-des-métriques-spécifiques).

   > Il est possible de ne définir qu'un seuil CRITIQUE, sans passer par le stade ALERTE (et vice-versa).

* **min/max** : Il ne s'agit pas de la valeur minimale/maximale constatée dans les données de performance, mais de la valeur minimale/maximale possible que peut prendre la métrique. Par exemple, le maximum d'une utilisation CPU sera toujours 100%. Pour certains plugins ou modes, ces valeurs peuvent être redéfinies [dans la macro EXTRAOPTIONS](#comment-définir-des-seuils-pour-des-métriques-spécifiques).

>    * Le min et le max seront les valeurs minimales/maximales sur l'axe des Y du graphique. Si aucun min et max ne sont définis, le graphique s'adaptera en hauteur en fonction de la plus petite et la plus grande valeur constatées. Redéfinir les valeurs min/max permettre de rendre l'affichage des graphes le plus pertinent possible : si plusieurs graphiques représentant le même service pour plusieurs hôtes sont affichés ensemble, définir le même min et max permet de représenter toutes les métriques à la même échelle.

Reviewer 2: ```suggestion
   * Le min et le max seront les valeurs minimales/maximales sur l'axe des Y du graphique. Si aucun min et max ne sont définis, le graphique s'adaptera en hauteur en fonction de la plus petite et la plus grande valeur constatées. Redéfinir les valeurs min/max permet de rendre l'affichage des graphes le plus pertinent possible : si plusieurs graphiques représentant le même service pour plusieurs hôtes sont affichés ensemble, définir le même min et max permet de représenter toutes les métriques à la même échelle.
```
